### PR TITLE
Add NICK command

### PR DESCRIPTION
**Pull request by vfurmane**

### CODE REVIEW

File: include/IRCPeerManager.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		IRCPeerManager &operator=(const IRCPeerManager &rhs);
+		~IRCPeerManager(void);
+
+		IRCPeer	&operator[](int fd);
+
+		void			add(int fd, struct sockaddr &addr);
+		void			remove(int fd);
+		IRCPeer			&get(int fd);
+		int				acceptConnection(void);
+		void			closeConnection(int fd);
```
Comment (theovgl) [inline include/IRCPeerManager.hpp:26]: Maybe it's not necessary ?

File: src/IRCPeerManager.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+void	IRCPeerManager::closeConnection(int fd)
+{
+#ifndef NDEBUG
+	std::cerr << "Closing connection on fd no " << fd << "..." << std::endl;
+	std::cerr << "IP address -> " << this->get(fd).getStrAddr() << std::endl;
+#endif
+	if (this->_peers.at(fd).close() == -1)
+		throw sysCallError("close", strerror(errno));
+	this->remove(fd);
+}
```
Comment (theovgl) [inline src/IRCPeerManager.cpp:74]: Same question as above :)

File: src/handlers.cpp [hunk 3]
```diff
@@ -7,12 +7,13 @@ void	parseInput(const std::string &input)
...
 {
+	(void)peer;
 	int		bytes_read;
 	char	buffer[MAX_READ + 1];
 	IRCPeer	*irc_peer = static_cast<IRCPeer *>(peer);
-	
+
 	bytes_read = recv(event->data.fd, buffer, MAX_READ, 0);
-	if (bytes_read <= 0)
+	if (bytes_read == 0)
```
Comment (theovgl) [inline src/handlers.cpp:16]: Merge error

**Review by theovgl — CHANGES_REQUESTED** [review]

File: include/IRCServer.hpp [hunk 4]
```diff
@@ -7,6 +7,9 @@
 # include "TCPServer.hpp"
 # include "handlers.hpp"
 
+class IRCPeerManager;
+# include "IRCPeerManager.hpp"
```
Comment (ppiques) [inline include/IRCServer.hpp:11]: useless line

**Review by ppiques — CHANGES_REQUESTED** [review]

File: include/IRCMessage.hpp [hunk 5]
```diff
@@ -17,10 +19,18 @@ class IRCMessage
 
 	void		parse();
 	void		execute();
-	
 	private:
+	
+	size_t	getArgsCount(void) const;
+	const std::string	*getArguments(void) const;
+	const std::string	&getInput(void) const;
```
Comment (vfurmane) [inline include/IRCMessage.hpp:26]: Why the fuck are the getters private you braindead animal ???????

**Review by ppiques — APPROVED** [review]

**Review by theovgl — APPROVED** [review]

File: src/handlers.cpp [hunk 6]
```diff
@@ -7,10 +7,11 @@ void	parseInput(const std::string &input)
 
 int	handleIRCMessage(TCPPeer *peer, epoll_event *event)
 {
+	(void)peer;
```
Comment (theovgl) [inline src/handlers.cpp:10]: :)

**Review by theovgl — CHANGES_REQUESTED** [review]

File: include/IRCMessage.hpp [hunk 7]
```diff
@@ -17,10 +19,18 @@ class IRCMessage
...
 
-	void	_parseArguments(std::string::const_iterator &it);
+	IRCPeer	&peer;
+
+	protected:
+
+	void				_parseArguments(std::string::const_iterator &it);
+	const std::string	&_updateInputFromFields(void);
```
Comment (ppiques) [inline include/IRCMessage.hpp:33]: will be obliterated by my PR anyway

File: include/IRCServer.hpp (hunk 4 above)
Comment (ppiques) [reply include/IRCServer.hpp:11]: still useless

File: src/handlers.cpp [hunk 8]
```diff
@@ -10,9 +10,9 @@ int	handleIRCMessage(TCPPeer *peer, epoll_event *event)
 	int		bytes_read;
 	char	buffer[MAX_READ + 1];
 	IRCPeer	*irc_peer = static_cast<IRCPeer *>(peer);
-	
+
 	bytes_read = recv(event->data.fd, buffer, MAX_READ, 0);
-	if (bytes_read <= 0)
+	if (bytes_read == 0)
```
Comment (theovgl) [inline src/handlers.cpp:15]: bon la c'est trop

**Review by theovgl — CHANGES_REQUESTED** [review]

File: src/handlers.cpp (hunk 8 above)
Comment (ppiques) [inline src/handlers.cpp:15]: Comme Theo il a dit

**Review by ppiques — CHANGES_REQUESTED** [review]
elle est longue un peu la PR la non ?

File: include/IRCMessage.hpp (hunk 7 above)
Comment (vfurmane) [reply include/IRCMessage.hpp:33]: Your PR will never happen you ignoble piece of garbage

**Review by theovgl — APPROVED** [review]

**Review by ppiques — APPROVED** [review]

File: Makefile [hunk 9]
```diff
@@ -1,5 +1,5 @@
 NAME = ircserv
-SRCS = $(addprefix src/, main.cpp Configuration.cpp TCPPeer.cpp IRCPeer.cpp TCPPeerManager.cpp TCPServer.cpp handlers.cpp IRCServer.cpp IRCMessage.cpp commands.cpp $(addprefix commands/, quit.cpp))
+SRCS = $(addprefix src/, main.cpp Configuration.cpp TCPPeer.cpp IRCPeer.cpp TCPPeerManager.cpp TCPPeerManager.cpp TCPServer.cpp handlers.cpp IRCServer.cpp IRCMessage.cpp commands.cpp $(addprefix commands/, nick quit.cpp))
```
Comment (ppiques) [inline Makefile]: TCPPeerManager.cpp en double

**Review by theovgl — APPROVED** [review]
🤔🤔🤔

**Review by ppiques — APPROVED** [review]
LGTM

**Pull request merged**